### PR TITLE
fix serializing UDP header struct and add test

### PR DESCRIPTION
**Pull request by fenollp**

### CODE REVIEW

File: src/udp.rs [hunk 1]
```diff
@@ -14,3 +14,22 @@ pub struct UdpHdr {
 impl UdpHdr {
     pub const LEN: usize = mem::size_of::<UdpHdr>();
 }
+
+#[test]
+#[cfg(feature = "serde")]
+#[cfg(feature = "bincode")]
```
Comment (vadorovsky) [inline src/udp.rs:20]: This is not going to work? There is no "bincode" feature. You just added bincode as a dev-dependency.

I think you could just write the test in the following way:

```rust
#[cfg(test)]
mod test {
    #[cfg(feature = "serde")]
    #[test]
    fn test_serialize() {
        // here goes your code
    }
}
```

**Review by vadorovsky — CHANGES_REQUESTED** [review]
Thanks! Just one comment.

File: src/udp.rs (hunk 1 above)
Comment (fenollp) [reply src/udp.rs]: It did work, thanks to the `dep:bincode` legacy notation :)
---
Done.

**Comment by fenollp** [comment]
Amended the test declaration. I also added a drive by docs fix to the ip module.

Feel free to merge #13 first and I'll rebase this.

**Review by vadorovsky — APPROVED** [review]
Thanks and sorry for late reaction

**Pull request merged**